packages/react-dom-bindings/src/server/ReactDOMServerFormatConfig.js

```diff
@@ -2743,9 +2886,32 @@ const arraySubsequentOpenBracket = stringToPrecomputedChunk(',[');
 const arrayInterstitial = stringToPrecomputedChunk(',');
 const arrayCloseBracket = stringToPrecomputedChunk(']');
 
+function writeStyleResourceObject(
+  destination: Destination,
+  str: string,
+  isScriptWriter: boolean,
```

Comment:
nit: this is critical for security reasons so maybe name this something to do with the embedding context rather than which feature is active. like `javascriptEmbedding` or `htmlAttributeEmbedding` and make it an enum and throw if an expected embedding is not provided.